mitosheet/src/mito/components/taskpanes/Merge/MergeTaskpane.tsx

```diff
@@ -106,15 +98,103 @@ export const getDefaultMergeParams = (sheetDataArray: SheetData[], _sheetIndexOn
+
+    const validMergeKeyColumnIDs = merge_key_column_ids.filter(([mergeKeyColumnIDOne, mergeKeyColumnIDTwo]) => {
+        return sheetDataOne.columnIDsMap[mergeKeyColumnIDOne] !== undefined && sheetDataTwo.columnIDsMap[mergeKeyColumnIDTwo] !== undefined
+    });
+
+    const validSelectedColumnIDsOne = selected_column_ids_one.filter((columnID) => {
+        return sheetDataOne.columnIDsMap[columnID] !== undefined
+    });
+
+    const validSelectedColumnIDsTwo = selected_column_ids_two.filter((columnID) => {
+        return sheetDataTwo.columnIDsMap[columnID] !== undefined
+    });
+
+    return {
+        ...params,
+        merge_key_column_ids: validMergeKeyColumnIDs,
+        selected_column_ids_one: validSelectedColumnIDsOne,
+        selected_column_ids_two: validSelectedColumnIDsTwo
+    }
+}
+
+const getEditMergeWarnings = (params: MergeParams, sheetDataOne: SheetData, sheetDataTwo: SheetData): {
+    mergeKeyWarnings?: string[],
+    selectedColumnsOneWarnings?: string,
+    selectedColumnsTwoWarnings?: string,
+} => {
+    const {
+        merge_key_column_ids,
+        selected_column_ids_one,
+        selected_column_ids_two 
+    } = params;
+
+    const invalidMergeKeys = merge_key_column_ids.filter(([mergeKeyColumnIDOne, mergeKeyColumnIDTwo]) => {
+        return sheetDataOne.columnIDsMap[mergeKeyColumnIDOne] === undefined || sheetDataTwo.columnIDsMap[mergeKeyColumnIDTwo] === undefined
+    });
+    const mergeKeyWarnings = invalidMergeKeys.map(([mergeKeyColumnIDOne, mergeKeyColumnIDTwo]) => {
+        const sheetOneColumnHeader = sheetDataOne.columnIDsMap[mergeKeyColumnIDOne];
+        const sheetTwoColumnHeader = sheetDataTwo.columnIDsMap[mergeKeyColumnIDTwo];
+        if (sheetOneColumnHeader === undefined && sheetTwoColumnHeader === undefined) {
+            return `The merge keys ${mergeKeyColumnIDOne} and “${mergeKeyColumnIDTwo}” were removed because they no longer exist in “${sheetDataOne.dfName}” and “${sheetDataTwo.dfName}”.`
+        } else if (sheetOneColumnHeader === undefined) {
+            return `The merge key pairing (${mergeKeyColumnIDOne}, ${mergeKeyColumnIDTwo}) was removed because “${mergeKeyColumnIDOne}” no longer exists in “${sheetDataOne.dfName}”.`
+        } else {
+            return `The merge key pairing (${mergeKeyColumnIDOne}, ${mergeKeyColumnIDTwo}) was removed because “${mergeKeyColumnIDTwo}” no longer exists in “${sheetDataTwo.dfName}”.`
+        }
+    });
 
+    const getSelectedColumnWarningString = (invalidSelectedColumns: ColumnID[], sheetData: SheetData): string | undefined => {
+        if (invalidSelectedColumns.length === 1) {
+            return `The column “${invalidSelectedColumns[0]}” was removed because it no longer exists in “${sheetData.dfName}”.`
+        } else if (invalidSelectedColumns.length > 1) {
+            return `The columns “${invalidSelectedColumns.join(', ')}” were removed because they no longer exist in “${sheetData.dfName}”.`
+        } else {
```

Comment:
TODO: check these with really long column names

Reply:
Ooo good point. Tested and it was a little funky but I pushed a CSS fix. This is the updated version, what do you think:
<img width="379" alt="image" src="https://github.com/mito-ds/mito/assets/6673460/d8dde135-5806-418b-83c9-d057058c4b16">